apps/web-mzima-client/src/app/settings/categories/categories.component.ts

```diff
@@ -62,14 +61,28 @@ export class CategoriesComponent {
         complete: () => {
           this.getCategories();
           this.selectedCategories = [];
-          this.notificationService.showError(
-            this.translate.instant('bulk_destroy_success_countless'),
-          );
+          this.notificationService.showError(messages.toast);
         },
       },
     );
   }
 
+  public confirmationMessages(): { title: string; description: string; toast: string } {
+    const isBulk = this.selectedCategories.length > 1;
+    const count = this.selectedCategories.length;
+    return {
+      title: this.getMessage('destroy_confirm', isBulk, count),
+      description: this.getMessage('destroy_confirm_desc', isBulk, count),
+      toast: this.getMessage('destroy_success', isBulk, count),
+    };
+  }
+
+  private getMessage(baseKey: string, isBulk: boolean = false, count?: number): string {
+    const prefix = isBulk ? 'bulk_' : '';
```

Comment:
Could we not use count to establish prefix?
eg. const prefix = count > 1 ? 'bulk_' : '';